crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -168,15 +168,19 @@ impl KeyValueStore for RocksDb {
             DATABASE_METRICS.write_meter.inc();
             DATABASE_METRICS.bytes_written.observe(value.len() as f64);
         }
+        // FIXME: This is a race condition. We should use a transaction.
         let prev = self.get(key, column)?;
+        // FIXME: This is a race condition. We should use a transaction.
         self.db
             .put_cf(&self.cf(column), key, value)
             .map_err(|e| DatabaseError::Other(e.into()))
             .map(|_| prev)
     }
 
     fn delete(&self, key: &[u8], column: Column) -> DatabaseResult<Option<Vec<u8>>> {
+        // FIXME: This is a race condition. We should use a transaction.
```

Comment:
This should have a github issue. I'm not sure how critical these are, but typically fixing a race condition should ba a high priority.

Reply:
It is also related to the todo above